File: src/main/java/data/Wow.java
package data;

import java.io.*;
import java.util.ArrayList;
import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class Wow {

    public static void main(String[] args) {

        try {
            ArrayList<Location> someData = getConvenienceStores();
        } catch (IOException e) {
            throw new RuntimeException(e);
        }

    }

    public static ArrayList<Location> getConvenienceStores() throws IOException {

        ArrayList<Location> result = new ArrayList<>();

        Pattern pattern = Pattern.compile(" showlocation\\(\"([\\w#.,'/@() \\-]+)\", \"([\\w#.,'/@() \\-]+)\", ([\\d.]+), ([\\d.]+)\\);");

        // load 7-11.txt
        try (BufferedReader br = new BufferedReader(new FileReader("data/7-11.txt"))) {
            String line = br.readLine();
            while (line != null) {
                Matcher matcher = pattern.matcher(line);
                if (matcher.matches()) {
                    Location location = new Location(matcher.group(1), Double.parseDouble(matcher.group(3)), Double.parseDouble(matcher.group(4)));
                    result.add(location);
                } else {
                    System.out.println("skill issue"); // replace with error or better text
                }
                line = br.readLine();
            }
        }
        catch (NumberFormatException exception) {
            exception.printStackTrace();
        } /* finally {
            // finally!
        } */
        // the buffered reader auto closes after this try

        return result;

    }

}

File: src/main/kotlin/data/Wow.kt
package data

import model.base.Location
import java.io.BufferedReader
import java.io.FileReader
import java.io.IOException
import java.net.URL
import java.util.regex.Pattern

@Throws(IOException::class)
fun sevenElevens(): ArrayList<Location> {
    val result = ArrayList<Location>()
    val pattern =
        Pattern.compile("showlocation\\(\"([\\w#.,'/@() \\-]+)\", \"([\\w#.,'/@() \\-]+)\", ([\\d.]+), ([\\d.]+)\\);")

    // load 7-11.txt
    try {
        BufferedReader(FileReader("data/7-11.txt")).use { br ->
            var line = br.readLine()
            while (line != null) {
                val matcher = pattern.matcher(line)
                if (matcher.matches()) {
                    val latitude = matcher.group(3).toDouble()
                    if (latitude == 0.0) {
                        // check to remove funny africa point
                        line = br.readLine()
                        continue
                    }
                    val longitude = matcher.group(4).toDouble()
                    val location = Location(matcher.group(1), latitude, longitude, getURL(latitude, longitude))
                    result.add(location)
                } // else?
                line = br.readLine()
            }
        }
    } catch (exception: NumberFormatException) {
        exception.printStackTrace()
    }
    // the buffered reader auto closes after this try
    return result
}

fun getURL(latitude: Double, longitude: Double): URL {
    return URL("https://google.com/maps/search/?api=1&query=%.9f,%.9f".format(latitude, longitude))
}